spec/acceptance/datasets_spec.rb

```diff
@@ -0,0 +1,215 @@
+    standard_request_options(:get, 'NEW (with invalid token)', :unauthorized, {expected_json_path: get_json_error_path(:sign_up)})
+  end
+
+  get '/datasets/new' do
+    standard_request_options(:get, 'NEW (as anonymous user)', :unauthorized, {expected_json_path: get_json_error_path(:sign_up)})
+  end
+
+  ################################
+  # SHOW
+  ################################
+
+  get '/datasets/:id' do
+    id_params
+    let(:id) { dataset.id }
+    let(:authentication_token) { admin_token }
+    standard_request_options(:get, 'SHOW (as admin)', :ok, {expected_json_path: 'data/name'})
+  end
+
+  get '/datasets/:id' do
+    id_params
+    let(:id) { dataset.id }
+    let(:authentication_token) { no_access_token }
+    standard_request_options(:get, 'SHOW (as non admin user)', :ok, {expected_json_path: 'data/name'})
+  end
+
+  get '/datasets/:id' do
+    id_params
+    let(:id) { dataset.id }
+    let(:authentication_token) { invalid_token }
+    standard_request_options(:get, 'SHOW (with invalid token)', :unauthorized, {expected_json_path: get_json_error_path(:sign_up)})
+  end
+
+  get '/datasets/:id' do
+    id_params
+    let(:id) { dataset.id }
+    standard_request_options(:get, 'SHOW (an anonymous user)', :unauthorized, {remove_auth: true, expected_json_path: get_json_error_path(:sign_up)})
+  end
+
+
+  ################################
+  # UPDATE
+  ################################
+
+  put '/datasets/:id' do
+    body_params
+    let(:id) { dataset.id }
+    let(:raw_post) { {dataset: post_attributes}.to_json }
+    let(:authentication_token) { admin_token }
+    standard_request_options(:put, 'UPDATE (as admin)', :ok, {expected_json_path: 'data/name', response_body_content: 'New Dataset name'})
+  end
+
+  put '/datasets/:id' do
+    body_params
+    let(:id) { dataset.id }
+    let(:raw_post) { {dataset: post_attributes}.to_json }
+    let(:authentication_token) { no_access_token }
+    standard_request_options(:put, 'UPDATE (as non admin user)', :forbidden, {expected_json_path: get_json_error_path(:permissions)})
+  end
+
+  put '/datasets/:id' do
+    body_params
+    let(:id) { dataset.id }
+    let(:raw_post) { {dataset: post_attributes}.to_json }
+    let(:authentication_token) { invalid_token }
+    standard_request_options(:put, 'UPDATE (with invalid token)', :unauthorized, {expected_json_path: get_json_error_path(:sign_up)})
+  end
+
+  put '/datasets/:id' do
+    body_params
+    let(:id) { dataset.id }
+    let(:raw_post) { {dataset: post_attributes}.to_json }
+    standard_request_options(:put, 'UPDATE (as anonymous user)', :unauthorized, {remove_auth: true, expected_json_path: get_json_error_path(:sign_in)})
+  end
+
+
+  ################################
+  # DESTROY
+  ################################
+
+  # Destroy is not implemented, so just one test for expected 404
+  delete '/datasets/:id' do
+    body_params
+    let(:id) { dataset.id }
+    let(:authentication_token) { admin_token }
```

Comment:
this is an ok exception to the above requirement of testing all well known users 👍